client/sections-middleware.js

```diff
@@ -0,0 +1,103 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import config from 'config';
+import page from 'page';
+import { find, filter, isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { activateNextLayoutFocus } from 'state/ui/layout-focus/actions';
+import * as LoadingError from 'layout/error';
+import * as controller from './controller/index.web';
+import { restoreLastSession } from 'lib/restore-last-path';
+import { hub as preloadHub } from 'sections-preload';
+import { switchCSS } from 'lib/i18n-utils/switch-locale';
+import { pathToRegExp } from './utils';
+
+import sections from './sections';
+
+const _loadedSections = {};
+
+function maybeLoadCSS( sectionName ) {
+	//eslint-disable-line no-unused-vars
+	const section = find( sections, { name: sectionName } );
+
+	if ( ! ( section && section.css ) ) {
+		return;
+	}
+
+	const url =
+		typeof document !== 'undefined' && document.documentElement.dir === 'rtl'
+			? section.css.urls.rtl
+			: section.css.urls.ltr;
+
+	switchCSS( 'section-css-' + section.css.id, url );
+}
+
+function preload( sectionName ) {
+	maybeLoadCSS( sectionName );
+	const filteredSections = filter( sections, { name: sectionName } );
+	if ( isEmpty( sections ) ) {
```

Comment:
I might be mistaken, but don't we want to check for emptiness of `filteredSections` here?

Reply:
nice catch!

this hasn't come up during testing because this should never really catch...why would we ever try to preload a section that doesn't exist.

We can refactor logic another time though, i'll fix that now